### PR TITLE
Added deprecated module to setup.py Vinay Vinod  <briarvinay@gmail.com>

### PR DESCRIPTION
**Pull request by Vinay-Vinod**
### Description
Added deprecated module to setup.py
 
### Issues Resolved
[BUG] deprecated library missing from install_requires list #315
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dhrubo-os** [comment]
lint is failing. Could you please check?

File: setup.py [hunk 1]
```diff
@@ -87,6 +87,7 @@
         "pandas>=1.5,<3",
         "matplotlib>=3.6.0,<4",
         "numpy>=1.24.0,<2",
+        "deprecated"
```
Comment (rawwar) [inline setup.py:90]: Can we pinning it to version latest(`1.2.14`) or a compatible

Comment (dhrubo-os) [reply setup.py:90]: +1 on this

File: setup.py [hunk 2]
```diff
@@ -87,6 +87,7 @@
         "pandas>=1.5,<3",
         "matplotlib>=3.6.0,<4",
         "numpy>=1.24.0,<2",
+        "deprecated>=1.2.14",
```
Comment (rawwar) [inline setup.py:90]: `deprecated>=1.2.14,<2.0.0` would be better.

**Comment by rawwar** [comment]
@dhrubo-os , can you approve the workflows to run. I think this can be merged if tests are passing.

**Comment by rawwar** [comment]
@dhrubo-os , I think the coverage report should be ignoring `setup.py` or any meta files related project. Apart from that, I think we can merge this PR

**Review by dhrubo-os — APPROVED** [review]

**Review by b4sjoo — APPROVED** [review]

**Pull request merged**